### PR TITLE
Fix the bug of using the same token for padding and eos

### PR DESCRIPTION
**Pull request by rao-ma**
Models trained with the original code cannot stop the generation in the decoding, leading to a high CER.

The original code uses eos_token as the pad_token. Then in the SpeechDataset iteration, tokens with the pad_token id are ignored in the loss calculation. Thus the model didn't learn how to generate EOS in the training. 

Therefore, it's a better practice to set padding_token to '<|finetune_right_pad_id|>'. This fix the issue and I was able to reproduce the ASR results using my updated codebase.

Please also refer to https://discuss.huggingface.co/t/how-to-set-the-pad-token-for-meta-llama-llama-3-models/103418/2

### CODE REVIEW

**Comment by robin1001** [comment]
Good Catch! Thanks
